converter/internal/otelcolconvert/converter_lokiexporter.go

```diff
@@ -0,0 +1,126 @@
+package otelcolconvert
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/grafana/agent/component/common/config"
+	"github.com/grafana/agent/component/loki/write"
+	"github.com/grafana/agent/component/otelcol/exporter/loki"
+	"github.com/grafana/agent/converter/diag"
+	"github.com/grafana/agent/converter/internal/common"
+	"github.com/grafana/river/rivertypes"
+	pconfig "github.com/prometheus/common/config"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configtls"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/lokiexporter"
+)
+
+func init() {
+	converters = append(converters, lokiExporterConverter{})
+}
+
+type lokiExporterConverter struct{}
+
+func (lokiExporterConverter) Factory() component.Factory {
+	return lokiexporter.NewFactory()
+}
+
+func (lokiExporterConverter) InputComponentName() string { return "otelcol.exporter.loki" }
+
+func (lokiExporterConverter) ConvertAndAppend(state *state, id component.InstanceID, cfg component.Config) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	label := state.FlowComponentLabel()
+
+	lokiWriteComponentID := []componentID{{
+		Name:  strings.Split("loki.write", "."),
+		Label: label,
+	}}
+
+	args1 := toOtelcolExporterLoki(lokiWriteComponentID)
+	block1 := common.NewBlockWithOverride([]string{"otelcol", "exporter", "loki"}, label, args1)
+
+	diags.Add(
+		diag.SeverityLevelInfo,
+		fmt.Sprintf("Converted %s into %s", stringifyInstanceID(id), stringifyBlock(block1)),
+	)
+
+	args2, err := toLokiWrite(label, cfg.(*lokiexporter.Config))
+	if err != nil {
+		diags.Add(
+			diag.SeverityLevelError,
+			fmt.Sprintf("could not build loki.write block: %s", err),
+		)
+	}
+	block2 := common.NewBlockWithOverride([]string{"loki", "write"}, label, args2)
+
+	diags.Add(
+		diag.SeverityLevelInfo,
+		fmt.Sprintf("Converted %s into %s", stringifyInstanceID(id), stringifyBlock(block2)),
+	)
+	diags.Add(
+		diag.SeverityLevelInfo,
+		fmt.Sprintf("Created a loki.write block with a best-effort conversion of the lokiexporter's confighttp, retry and queue configuration settings. You may want to double check the converted configuration as most fields do not have a 1:1 match"),
+	)
+
+	state.Body().AppendBlock(block1)
+	state.Body().AppendBlock(block2)
+	return diags
+}
+
+func toOtelcolExporterLoki(ids []componentID) *loki.Arguments {
+	return &loki.Arguments{
+		ForwardTo: toTokenizedLogsReceivers(ids),
+	}
+}
+
+func toLokiWrite(name string, cfg *lokiexporter.Config) (*write.Arguments, error) {
+	// Defaults for MaxStreams and WAL should be handled on the Flow side.
+	res := &write.Arguments{}
+
+	if cfg.Endpoint != "" {
+		// TODO(@tpaschalis) Wire in auth from auth extension.
```

Comment:
(Just a note) 

😬 Handling auth here will be tricky, because we can't pass `otelcol.auth.basic` to `loki.write`; we'll have to actually probe the auth extension that lokiexporter was using and copy settings from the definition of the auth extension into `loki.write`.